### DISCUSSION TITLE
VX3 Anschluss Canbus

### DISCUSSION THREAD

**Discussion by Figo81** [open3e/open3e#102] [Q&A] [answered]
Hallo, ich habe den Canbus am VX3 abgegriffen. Dafür habe ich das Netzwerkkabel an der eingezeichneten Stelle mit angeschlossen. Ich habe auch Daten empfangen.
Allerdings ist einige Zeit später das Display ausgefallen und ein Techniker musste das Display tauschen. Ich vermute, dass es im Zusammenhang mit dem Mehrfachanschluss an einem Stecker zu einem Problem kam. Wie habt ihr das am VX3 gelöst? Einen zweiten Bus-Anschluss finde ich nicht.

![PXL_20240515_150432384~2](https://github.com/open3e/open3e/assets/76645300/ec4b1f25-36f2-4ebf-ae1e-fb5469c48031)

**Comment by philippoo66** [comment]
ich denke nicht, dass ein Ausfall des Displays in einem kausalen Zusammenhang mit 'Mehrfach-Anschluss' steht (höchstens in einem zeitlichen). Bei den Geräten mit 2 Steckern passiert intern nichts anderes:

![grafik](https://github.com/open3e/open3e/assets/122479122/d05f2059-0136-4fc0-88fb-672b8cca448e)

Was ganz eventuell problematisch sein _könnte_, ist das Fehlen der Masseverbindung. Mit oder ohne ist es immer wichtig, das gesamte Ground-Schema zu betrachten. Ein isolierter Transceiver sollte immer mit einer Masse-Verbindung angeschlossen werden. Bei einem nicht-isolierten ist darauf zu achten, dass die ganzen Grounds auf einem Potential liegen...

Was ist denn am Display kaputt gegangen? funktionierte es komplett nicht mehr oder nur die CAN Kommunikation?

ps: wenn du schon verdrillte Leiter hast, solltest du CAN_HI und CAN_LO paarweise miteinander verdrillen, nicht hi mit hi und lo mit lo. Der Grund dafür ist die Kopensation von Einstreuungen bei verdrillten Paaren.

**Reply by Figo81** [reply]
Das Display ist komplett ausgefallen und das System funktionierte dann auch nicht mehr. Was genau daran defekt war, konnte man nicht erkennen.
Also dürfte ein Anschluss von Can-L und Can-H kein Problem sein. Sollte ich dann Ground an eine Erdung anschließen? Oder besser komplett weglassen?

Die auf dem Bild verdrillten Kabel sind schon vom Solarteur so installiert worden.  Ich habe mein Netzwerkkabel auch paarweise verdrillt so angeschlossen (dazugeklemmt). Auf der anderen Seite des Kabels aber dann auch paarweise jeweils an Can-L und Can-H (also 2 auf 2 jeweils). Sollte ich das mischen? Aber das würde doch die Signale verfälschen?

**Reply by philippoo66** [reply]
das hier ist der Hintergrund:

![grafik](https://github.com/open3e/open3e/assets/122479122/33d55250-15b3-4f12-bdc6-448f2c486e92)

oben bewirkt die Einstahlung einen überlagerten Unterschied zwischen CAN_Hi und CAN_Lo (also ein eingestrahltes Störsignal). Weil unten die Orientierung der aufeinander folgenden Schleifen jeweis andersrum ist, löscht die Einstahlung in die eine Schleife die von der nebenliegenden wieder aus. [Hier](https://creafield.ch/leitungen-verdrillen-oder-abschirmen-was-ist-besser/) ist die ganze Seite zu dem Thema.

Wenn du aber nur Hi mit Hi und Lo mit Lo verdrillst, liegen die Bündel immer noch nebeneinander und spannen -wie oben- _eine_ große Fläche zwischen Hi und Lo auf (nicht viele invertiert ausgerichtete) .

Wenn an den 4 Adern auf deinem Bild 2 Geräte hängen (Gerät und Adapter), solltest du bsw.

blau und braun ->  Hi 
blau/weiss und braun/weiss -> Lo

belegen (natürlich auf beiden Seiten), weil gewöhnlich blau und blau/weiss miteinander verdrillt sind und bei braun das Gleiche. Dann ein Gerät an blau+blau/weiss und das andere an braun+braun/weiss. 

Eigentlich sollten dann aber die Geräte an dem anderen Ende des Kabels verbunden werden (das spart Buslänge). 

Wenn das Kabel nicht 100 Meter lang ist und einen einigermassen Querschnitt hat, ist es besser, nur ein Adernpaar zu benutzen, weil die parallel geführten Kabel auch immer einen Kondensator bilden... Wenn der doppelte Querschnitt unbedingt benötigt wird, dann so machen wie oben beschrieben.

**Reply by philippoo66** [reply]
> Die auf dem Bild verdrillten Kabel sind schon vom Solarteur so installiert worden. 

Ein Solateur ist eben ein Solateur, und dieser hat offensichtlich wenig Ahnung von Signal-/Nachrichtentechnik.

> Sollte ich dann Ground an eine Erdung anschließen?

das ist gewöhnlich keine gute Idee! Wie gesagt - es gilt das gesamte System zu betrachten (in Bezug auf Grounds). Wichtig ist auch, das man auch hier keine Schleifen mit großen Flächen zum Einstrahlen baut. Für Signale ist eine sternförmiger Ground deswegen immer gut. Wenn aber irgendwo was isoliert ist, darf die Isolierung natürlich nicht durch Verbinden der Grounds (bsw. per Erdung) wieder überbrückt werden!

Eigentlich ist es optimal, den Gnd im Signalkabel mitzuführen, dann hat man die kleinste Fläche zum Einstrahlen von Störungen. Wenn aber (wie beispielsweise wohl bei dem Energiezähler der Fall) die Geräte-Grounds schon mit Erde verbunden sind, nutzt das auch wenig.

Das Thema ist komplex, und ohne eine Skizze der Geräte, ohne das Wissen, bei welchem CAN_Gnd==PE ist und wo ein isolierter CAN Anschluss vorliegt, kann keine Aussage getrofffen werden, wie jetzt wo welcher Gnd anzuschliessen ist. 

Ich habe dazu (ich glaube) im Viessmann Thread mal eine lange Abhandlunng geschrieben, ggf. mal da nach suchen. [Hier](https://github.com/open3e/open3e/wiki/020-Inbetriebnahme-CAN-Adapter-am-Raspberry#e380) eine Empfehlung zum Anschluss des empfohlenen isolierten Adapters...

**Reply by Figo81** [reply]
Ich danke dir erstmal für deine unglaublich ausführliche Antwort. Ich muss gestehen, dass ich da nicht vollends durchblicke. Das Kabel zum PC ist nur einen halben Meter lang. Macht es da etwas aus, wenn ich das nicht wie von dir beschrieben neu verdrille? Ich traue mich da nicht dran und will die Anlage nicht schrotten. (Das Kabel vom Wechselrichter zum Zählerkasten ist auch nur etwa 1m lang.)
Kann ich Gnd vom PC kommend dann einfach weglassen und nicht anschließen?

**Reply by philippoo66** [reply, marked as answer]
> Macht es da etwas aus, wenn ich das nicht wie von dir beschrieben neu verdrille?

gib mal 
```
ip -details -statistics link show can0
```
ein und schau was dabei raus kommt. Wenn das 'sauber' ist, da also 0 Fehler und 0 verworfene Pakete stehen, läuft die Kommunikation sauber, und aller Theorie zum Trotz kannst du deine Verkabelung so lassen wie sie ist ;-) Wenn da eine relevante Anzahl schief Gegangenes drin steht, sollten wir uns noch mal näher mit der Kabelei beschäftigen. 

Im Prinzip gilt das auch im Bezug auf die Masse, weil Probieren geht halt über studieren. Trotzdem noch mal grundsätzlich: Wenn du einen nicht isolierten CAN Adapter benutzt, würde ich die Masse ohne vorherige Potentialbetrachtung/-messung nicht verbinden, wenn der Adapter isoliert ist, dann ist es eigentlich schon geboten, die Masse anzuschliessen. 

Aber wie gesagt - guck die detaillierte Statistik der Kommunikation an und entscheide dann. Wenn du dir bei der Interpretation nicht sicher bist, poste den Output hier und wir gucken gemeinsam drüber.

Welchen CAN-'PC' Adapter benutzt du eigentlich?

**Reply by philippoo66** [reply]
ps. wenn nicht schon geschehen, guck dir das sehr kurze Video zum Verdrillen an: https://creafield.ch/wp-content/uploads/2024/03/Leitungen-verdrillen-cut.mp4 das ist wirklich genau so kurz wie einleuchtend

**Reply by Figo81** [reply]
Super! Danke für das Video und die Tipps.
Habe jetzt ohne Gnd angeschlossen und IP durchlaufen lassen. Waren keine Fehler. System läuft soweit wieder. Hoffentlich ohne Ausfall des Displays jetzt. *fingerscrossed*
Ich habe den Adapter gekauft, der in der Wiki auch empfohlen wurde.
Wünsche dir noch ein schönes Wochenende.

**Reply by MyHomeMyData** [reply]
Wenn es funktioniert und in der Statistik keine Fehler auftauchen: Nichts mehr ändern.

Sollten die Fehlerzähler aber doch nach oben gehen: Es ist doch nicht notwendig, Drähte zu verdrillen. Du kannst die vorhandenen, verdrillten Adern nutzen:
* Alle Kabel auf beiden Seiten ausklemmen
* EIN verdrilltes Paar für CAN_L und CAN_H nutzen, also z.B. CAN_H -> blau, CAN_L -> blau-weiß
* Das andere Adernpaar wird für die Verbindung VX3 <-> USB2CAN-Adapter nicht gebraucht. Abzwicken oder isolieren

**Reply by philippoo66** [reply]
ps. evtl brauchst du auch eine Ader von dem freiwerdenden Paar als CAN_Gnd. also nicht abzwicken!
